src/mlpack/methods/ann/models/yolov3/yolov3_tiny.hpp

```diff
@@ -0,0 +1,158 @@
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_MODELS_YOLOV3_TINY_HPP
+#define MLPACK_METHODS_ANN_MODELS_YOLOV3_TINY_HPP
+
+#include <mlpack/prereqs.hpp>
+
+#include <mlpack/methods/ann/dag_network.hpp>
+#include <mlpack/methods/ann/loss_functions/loss_functions.hpp>
+#include <mlpack/methods/ann/init_rules/init_rules.hpp>
+
+#include <mlpack/methods/ann/layer/layer_types.hpp>
+#include <mlpack/methods/ann/models/yolov3/yolov3_layer.hpp>
+
+namespace mlpack {
+/**
+ * YOLOv3-tiny is a small one-stage object detection model.
+ *
+ * The input to the model must be a square image. Look at image_letterbox.hpp
+ * to preprocess images before inference.
+ *
+ * The output of the model is a matrix. The rows are data points per
+ * bounding box (including x, y, w, h, objectness, and classifications).
+ * The columns represent each bounding box.
+ *
+ * For more information, refer to the following paper:
+ *
+ * @code
+ * @article{yolov3,
+ *   title     ={YOLOv3: An Incremental Improvement},
+ *   author    ={Redmon, Joseph and Farhadi, Ali},
+ *   journal   = {arXiv},
+ *   year      ={2018}
+ * }
+ * @endcode
+ *
+ */
+template <typename OutputLayerType = EmptyLoss,
+          typename InitializationRuleType = RandomInitialization,
+          typename MatType = arma::mat>
+class YOLOv3Tiny
+{
+ public:
+  // Helper types.
+  using ModelType =
+    DAGNetwork<OutputLayerType, InitializationRuleType, MatType>;
+  using Type = typename MatType::elem_type;
+  using CubeType = typename GetCubeType<MatType>::type;
+
+  YOLOv3Tiny() { /* Nothing to do. */ }
+
+  /**
+   * Create the YOLOv3Tiny model.
+   *
+   * @param imgSize The width and height of input images. Pretrained weights
+       used 416.
+   * @param numClasses The number of output classes. Pretrained weights were
+       trained on COCO which has 80 classes.
+   * @param predictionsPerCell Each YOLO layer predicts `predictionsPerCell`
+       boxes per grid cell. Pretrained weights use 3.
+   * @param anchors Vector of anchor width and heights. Formatted as
+      [w0, h0, w1, h1, ... ]. Each anchors is a [w, h] pair. There must be
+      predictionsPerCell * 2 anchors, since YOLOv3Tiny has two output layers.
+      Therefore, anchors.size() must be predictionsPerCell * 4.
+   */
+  YOLOv3Tiny(const size_t imgSize,
+             const size_t numClasses,
+             const size_t predictionsPerCell,
+             const std::vector<Type>& anchors);
+
+  ~YOLOv3Tiny() { /* Nothing to do. */ }
+
+  /**
+   * Returns the graph representation of the model.
+   */
+  ModelType& Model() { return model; }
+
+  /**
+   * Ordinary feed forward pass of the network.
+   *
+   * @param input Input data used for evaluating the specified function.
+      The input matrix dimensions should be (imgSize * imgSize, batchSize).
+   * @param output Resulting bounding boxes.
+   */
+  void Predict(const MatType& input, MatType& output)
```

Comment:
So, you are planning in a future PR to add/change this functionality so that instead of returning a matrix, it returns e.g. a vector of bounding box objects or similar?